src/blob/index/range.rs

```diff
@@ -0,0 +1,60 @@
+use super::prelude::*;
+
+/// NOTE: le and lt operations are written for big-endian format of keys
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub(crate) struct RangeFilter {
+    min: Vec<u8>,
+    max: Vec<u8>,
+    initialized: bool,
+}
+
+impl RangeFilter {
+    pub(crate) fn new() -> Self {
+        Self {
+            initialized: false,
+            ..Default::default()
+        }
+    }
+
+    pub(crate) fn add(&mut self, key: impl AsRef<[u8]>) {
+        let key_slice = key.as_ref();
+        if !self.initialized {
+            self.min = key_slice.to_vec();
+            self.max = key_slice.to_vec();
+            self.initialized = true;
+        } else if Self::lt(key_slice, &self.min) {
+            self.min = key_slice.to_vec()
+        } else if Self::lt(&self.max, key_slice) {
+            self.max = key_slice.to_vec()
+        }
```

Comment:
Is there a guarantee, that this code will be called from a single thread? Because it is not thread safe

Reply:
Yes, there is.
Simple answer: Rust allows only 1 mutable ref **or** N shared (immutable) refs (and doesn't allow to have it simultanously) and if you don't use `unsafe` then this guarantee won't be violated (and we don't). 
More strict answer: `Safe` struct which contains all blobs is under `RWLock`, so this is a guarantee.